lib/parser/ast/node.rb

```diff
@@ -34,6 +34,21 @@ def assign_properties(properties)
           @location = location
         end
       end
+      
+      ##
+      # Produces a machine-readable S-Expression.
+      #
+      def to_json
+        children_json = children.map do |child|
+          if child.is_a?(Node)
+            child.to_json
+          else
+            child
+          end
+        end
+
+        [type.to_s, *children_json]
+      end
```

Comment:
This should go into the `ast` gem, I think.

Reply:
I was thinking the same. I'll submit a PR there. Thanks!

Reply:
(Do you have quick feedback on whether this is roughly the right code? I don't write a ton of ruby)

Reply:
Code looks OK.

Reply:
Thanks. Posted in https://github.com/whitequark/ast/pull/20, will remove this commit